trends/src/main/kotlin/net/thechance/trends/service/ReelsService.kt

```diff
@@ -120,6 +126,53 @@ class ReelsService(
         return reelsRepository.save(updatedReel)
     }
 
+    @Transactional
+    fun incrementViewCount(reelId: UUID, userId: UUID) {
+        if (!reelViewRepository.existsByReelIdAndUserId(reelId, userId)) {
+            reelViewRepository.save(ReelView(reelId = reelId, userId = userId))
+
+            reelsRepository.findById(reelId).ifPresent { reel ->
+                reelsRepository.save(reel.copy(viewsCount = reel.viewsCount + 1))
+            }
+        }
+    }
+
+    @Transactional
+    fun toggleLike(reelId: UUID, currentUserId: UUID) {
+        val reel = reelsRepository.findById(reelId)
+            .orElseThrow { ReelNotFoundException() }
+
+        val isCurrentlyLiked = reelLikeRepository.existsByReelIdAndUserId(reelId, currentUserId)
+
+        if (isCurrentlyLiked) {
+            unlikeReel(reelId, currentUserId, reel)
+        } else {
+            likeReel(reelId, currentUserId, reel)
+        }
+    }
+
+    private fun likeReel(reelId: UUID, userId: UUID, reel: Reel) {
+        reelLikeRepository.save(ReelLike(reelId = reelId, userId = userId))
+        reelsRepository.save(reel.copy(likesCount = reel.likesCount + 1))
+    }
+
+    private fun unlikeReel(reelId: UUID, userId: UUID, reel: Reel) {
+        reelLikeRepository.deleteByReelIdAndUserId(reelId, userId)
+        reelsRepository.save(reel.copy(likesCount = maxOf(0,  - 1)))
```

Comment:
I think this should be:

```suggestion
reelsRepository.save(reel.copy(likesCount = maxOf(0, reel.likesCount - 1)))
```

Reply:
Yes, nice catch